### PR TITLE
Fixed bound to 4. 

### LINKED ISSUE #90: [auto-prove] No proof found while OK with Wu's prover
Automated prover stops with no proof found after 3s while [Wu's prover](https://github.com/wujuihsuan2016/LL_prover) is answering immediately:
```
⊢ X^⊗X^, X, ?(X^⊗(X⅋X))
```

### PR DESCRIPTION
**Pull request by etiennecallies**
Fixes #90

I try several possibilities of bound, 4 is required to prove all proofs in test. 5 is slower, 6 is too slow for certain sequents.

We probably have to be smarter in the future.

What do you think @olaure01 ?

### CODE REVIEW

**Comment by olaure01** [comment]
The alternative would be have an external loop calling the prover with 0, 1, 2, 3, etc. until time-out is reached (or starting with bigger initial value than 0, but this might miss shorter proofs if I remember well).

File: focused_proof.ml [hunk 1]
```diff
@@ -296,6 +295,13 @@ let rec prove sequent select_d2 max_d2 ttl =
               None
       | _ -> None
 
+let rec prove_with_increasing_bound focused_sequent exponential_bound ttl =
+    match prove focused_sequent [] exponential_bound ttl with
```
Comment (olaure01) [inline focused_proof.ml:299]:
```suggestion
    has_reached_exponential_bound := false;
    match prove focused_sequent [] exponential_bound ttl with
```

File: focused_proof.ml [hunk 2]
```diff
@@ -305,10 +311,8 @@ let rec prove sequent select_d2 max_d2 ttl =
 let prove_focused_sequent focused_sequent =
   let max_execution_time_in_seconds = 3. in
   let ttl = Sys.time () +. max_execution_time_in_seconds in
-  (* max_d2 is set as -1: bound will be increasingly limited *)
-  try match prove focused_sequent [] (-1) ttl with
-    | None -> (Some false, None)
-    | Some proof -> (Some true, Some proof)
+  has_reached_exponential_bound := false;
```
Comment (olaure01) [inline focused_proof.ml:314]:
```suggestion
```

File: focused_proof.ml (hunk 1 above)
Comment (olaure01) [reply focused_proof.ml:299]: `has_reached_exponential_bound` should be reset at each increment of `exponential_bound`

**Review by olaure01 — CHANGES_REQUESTED** [review]

File: focused_proof.ml (hunk 1 above)
Comment (etiennecallies) [reply focused_proof.ml:299]: Of course 👍🏻

**Pull request merged**